qiskit/optimization/problems/quadratic_program.py

```diff
@@ -152,36 +163,163 @@ def _add_variable(self,
+                                                   formatter, keys, initial_structure,
+                                                   update_function)
+
+    def _create_var_struct_update_maps(self,
+                                       lowerbound: Union[float, int],
+                                       upperbound: Union[float, int],
+                                       vartype: VarType,
+                                       name: str,
+                                       formatter: str,
+                                       keys: Union[int, Sequence],
+                                       initial_structure: Union[List[Variable],
+                                                                Dict[str, Variable]],
+                                       update_function) -> Union[List[Variable],
+                                                                 Dict[str, Variable]]:
+        acc_structure = initial_structure
+        if isinstance(keys, Sequence):
+            for key in keys:
+                indexed_name = name + formatter.format(str(key))
+                if indexed_name in self._variables_index:
+                    raise QiskitOptimizationError("Variable name already exists: {}"
+                                                  .format(indexed_name))
+                acc_structure = update_function(acc_structure,
+                                                self._create_var_update_maps(lowerbound,
+                                                                             upperbound,
+                                                                             vartype,
+                                                                             indexed_name))
+        else:
+            k = self.get_num_vars()
+            for _ in range(keys):
+                while name + formatter.format(k) in self._variables_index:
+                    k += 1
+                indexed_name = name + formatter.format(k)
+                acc_structure = update_function(acc_structure,
+                                                self._create_var_update_maps(lowerbound,
+                                                                             upperbound,
+                                                                             vartype,
+                                                                             indexed_name))
+        return acc_structure
+
+    def _create_var_update_maps(self,
+                                lowerbound: Union[float, int],
+                                upperbound: Union[float, int],
+                                vartype: VarType,
+                                name: str) -> Variable:
+        self.variables_index[name] = len(self.variables)
+        variable = Variable(self, name, lowerbound, upperbound, vartype)
+        self.variables.append(variable)
+        return variable
+
+    def var_dict(self,
```

Comment:
`var_dict` and `var_list` should not be public methods. `{binary,continuous,integer}_var_{list,dict}` is sufficient.

Reply:
See my PR too.
https://github.com/JosDenmark/qiskit-aqua/pull/1